src/game/Spell.cpp

```diff
@@ -4256,7 +4261,7 @@ SpellCastResult Spell::CheckCast(bool strict)
             for (Unit::SpellAuraHolderMap::iterator iter = Auras.begin(); iter != Auras.end(); ++iter)
             {
                 spell = iter->second->GetSpellProto();
-                if (spell->Dispel == mechanic)
+                if (spell->Dispel == mechanic || (m_spellInfo->Id == 24406 && (spell->Dispel == DISPEL_MAGIC || spell->Dispel == DISPEL_CURSE || spell->Dispel == DISPEL_DISEASE || spell->Dispel == DISPEL_POISON)))
```

Comment:
Yeah, this right here does not look well... It looks a bit like a hardcoded hackfix, wouldn't you say? =/ - Not saying there's any prettier way for us to check it by what we know currently, but still..